src/artifacts.ml

```diff
@@ -1,83 +1,120 @@
 open! Stdune
 open Import
 
-type t =
-  { context     : Context.t
-  ; local_bins  : Path.t String.Map.t lazy_t
-  ; public_libs : Lib.DB.t
-  }
+module Bin = struct
 
-let create (context : Context.t) ~public_libs =
-  let bin_dir = Config.local_install_bin_dir ~context:context.name in
-  let local_bins =
-    lazy (
-      Build_system.targets_of ~dir:bin_dir
-      |> Path.Set.fold ~init:String.Map.empty ~f:(fun path acc ->
-        let name = Filename.basename (Path.to_string path) in
-        (* The keys in the map are the executable names
-         * without the .exe, even on Windows. *)
-        let key =
-          if Sys.win32 then
-            Option.value ~default:name
-              (String.drop_suffix name ~suffix:".exe")
-          else
-            name
-        in
-        String.Map.add acc key path)
-    )
-  in
-  { context
-  ; local_bins
-  ; public_libs
+  module Partial = struct
+    type t = Path.t String.Map.t
+
+    let empty = String.Map.empty
+
+    let add_binaries t ~dir bindings =
+      List.fold_left bindings ~init:t
+        ~f:(fun acc fb ->
+          let path = File_binding.Expanded.dst_path fb
+                       ~dir:(Utils.local_bin dir) in
+          String.Map.add acc (Path.basename path) path)
+
+
+    let merge ~shadowing:x ~shadowed:y =
+      String.Map.merge x y ~f:(fun _key x y ->
+        match x with
+        | Some x -> Some x
+        | None ->
+          y)
+
+  end
+
+  type t = {
+    context : Context.t;
+    bin : Partial.t lazy_t;
```

Comment:
Looking at the code, this `lazy` is no longer justified, we should get rid of it. It was probably introduced in a previous version of the code that was doing more.

Reply:
Indeed. In fact the whole Partial module no longer needs to be exposed. I think the idea behind `Partial` was to have `Env_node`s carry it around, but that didn't work out.

Reply:
Seems good. BTW, I removed the `Partial` sub-module entirely. It didn't seem relevant anymore